src/FreqWindow.h

```diff
@@ -137,8 +141,7 @@ class FrequencyPlotDialog final : public wxDialogWrapper,
 
 
    double mRate;
-   size_t mDataLen;
-   Floats mData;
+   std::vector<ChannelGroupSampleView> mChannelGroupSampleViews;
```

Comment:
I see a new member variable in this dialog, which is non-modal and so may live a long time between button clicks.

That makes me want to examine its uses and see if instead local variables are used, or if that's not easy, then to be sure that the vector is cleared at appropriate times when the data are no longer needed.

Reply:
> see if instead local variables are used

Not easy: the audio is acquired in `GetAudio`, and read in another method: `Recalc`.

> be sure that the vector is cleared at appropriate times when the data are no longer needed

See `OnCloseButton()`.

This PR does not change the times when memory is acquired or released.

Reply:
I'm not persuaded by this either.

After a closer look, I think this:
- Only a local variable in `Recalc` is necessary.
- `GetAudio` can return the value instead of changing a member variable.
- `Recalc` can call `GetAudio`.  Both existing calls to `GetAudio` can be eliminated with this change.
- `DrawPlot` and `PlotPaint` only use the number `GetNumSamples(mChannelGroupSampleViews)` so instead, just store that number as a member variable each time `Recalc` is called.

Reply:
I think you are correct that you make the lifetime of sample data just the same as in the previous version.

Which means, the previous version didn’t handle memory as efficiently as it might have and so we can improve on it now.

With closer study you can find that the “analyst” object has all the derived data necessary to draw the plot, and the size of that is proportional only to the analysis window size (or number of frequency bins) and independent of the length of the selection, which could be large.

Reply:
I see three stages that matter to our decision here:
1. Selection sample read-out (`GetAudio`)
2. Calculation of spectrum (`Recalc`)
3. Display of result (`Show`)

`GetAudio` itself calls `Recalc`, but `Recalc` can also be called independently, if there is a change of analysis parameter (window size, spectrum type, ...). The user may keep the analysis window open and change the track data, which is a nice user experience. So the time-domain audio must be kept in memory for the lifetime of the window, we cannot assume that it won't change and we can re-read it.